source/service_test.go

```diff
@@ -267,13 +281,32 @@ func testServiceEndpoints(t *testing.T) {
 			},
 		},
 		{
+			"services annotated with legacy molecule annotations return an endpoint in compatibility mode",
+			"",
+			"testing",
+			"foo",
+			"molecule",
+			"",
+			map[string]string{
+				"dns": "route53",
+			},
+			map[string]string{
+				"domainName": "foo.example.org.",
```

Comment:
It works in both ways. I'm not going to add a test as it's mostly tested as part of `NewEndpoint()`